src/benchmarks/real-world/Roslyn/Program.cs

```diff
@@ -3,43 +3,28 @@
 using System.IO;
-using System.Threading.Tasks;
-using BenchmarkDotNet.Configs;
 using BenchmarkDotNet.Running;
 using BenchmarkDotNet.Extensions;
 using BenchmarkDotNet.Jobs;
+using CompilerBenchmarks;
 
-namespace CompilerBenchmarks
+string cscSourceDownloadLink = "https://github.com/dotnet/roslyn/releases/download/perf-assets-v2/CodeAnalysisReproWithAnalyzers.zip";
+string sourceDownloadDir = Path.Combine(AppContext.BaseDirectory, "roslynSource");
+var sourceDir = Path.Combine(sourceDownloadDir, "CodeAnalysisReproWithAnalyzers");
+if (!Directory.Exists(sourceDir))
 {
-    public class Program
-    {
-        public static async Task<int> Main(string[] args)
-        {
-            await Setup();
-
-            return BenchmarkSwitcher
-                .FromAssembly(typeof(Program).Assembly)
-                .Run(args, RecommendedConfig.Create(
-                               artifactsPath: new DirectoryInfo(Path.Combine(Path.GetDirectoryName(typeof(Program).Assembly.Location),
-                                                                             "BenchmarkDotNet.Artifacts")),
-                               mandatoryCategories: ImmutableHashSet.Create("Roslyn"),
-                               job: Job.Default.WithMaxRelativeError(0.01)))
-                .ToExitCode();
-        }
+    await FileTasks.DownloadAndUnzip(cscSourceDownloadLink, sourceDownloadDir);
+}
 
-        private static async Task Setup()
-        {
-            string cscSourceDownloadLink = "https://github.com/dotnet/roslyn/releases/download/perf-assets-v1/CodeAnalysisReproWithAnalyzers.zip";
-            string sourceDownloadDir = Path.Combine(AppContext.BaseDirectory, "roslynSource");
-            var sourceDir = Path.Combine(sourceDownloadDir, "CodeAnalysisReproWithAnalyzers");
-            if (!Directory.Exists(sourceDir))
-            {
-                await FileTasks.DownloadAndUnzip(cscSourceDownloadLink, sourceDownloadDir);
-            }
+// Benchmark.NET creates a new process to run the benchmark, so the easiest way
+// to communicate information is pass by environment variable
+Environment.SetEnvironmentVariable(Helpers.TestProjectEnvVarName, sourceDir);
 
-            // Benchmark.NET creates a new process to run the benchmark, so the easiest way
-            // to communicate information is pass by environment variable
-            Environment.SetEnvironmentVariable(Helpers.TestProjectEnvVarName, sourceDir);
-        }
-    }
-}
+return BenchmarkSwitcher
+    .FromAssembly(typeof(Helpers).Assembly)
+    .Run(args, RecommendedConfig.Create(
+                   artifactsPath: new DirectoryInfo(Path.Combine(Path.GetDirectoryName(typeof(Helpers).Assembly.Location),
+                                                                 "BenchmarkDotNet.Artifacts")),
+                   mandatoryCategories: ImmutableHashSet.Create("Roslyn"),
+                   job: Job.Default.WithMaxRelativeError(0.01)))
+    .ToExitCode();
```

Comment:
to be honest with you I am not sure if I like the new refactoring. I just find it odd to see `Program.Main` missing and it's also harder for me as a reviewer to say what exactly has changed

@billwert what do you think?

Reply:
I split this PR into 2 commits to make it easier to review: 1st commit is the new change, second is refactor to top-level-statements.

Reply:
I'm also not personally a fan of top level statements yet. I think for scratch programs or examples (in docs) they are great, but I'm not there yet for checked in code.

That said, with the commit split it's easy enough to see the actual diff, and this is a part of the code the Roslyn team basically owns at this point, so I don't have a strong objection.

Reply:
> I split this PR into 2 commits to make it easier to review

Thanks for doing that and apologies for not discovering that myself.

> with the commit split it's easy enough to see the actual diff, and this is a part of the code the Roslyn team basically owns at this point, so I don't have a strong objection

👍